Content.Shared/_Impstation/CosmicCult/SharedMonumentSystem.cs

```diff
@@ -0,0 +1,163 @@
+        _uiSystem.CloseUi(ent.Owner, MonumentKey.Key);
+    }
+    private void OnGlyphRemove(Entity<MonumentComponent> ent, ref GlyphRemovedMessage args)
+    {
+        if (ent.Comp.CurrentGlyph is not null) QueueDel(ent.Comp.CurrentGlyph);
+        _uiSystem.SetUiState(ent.Owner, MonumentKey.Key, GenerateBuiState(ent.Comp));
+        _uiSystem.CloseUi(ent.Owner, MonumentKey.Key);
+    }
+
+    private void OnInfluenceSelected(Entity<MonumentComponent> ent, ref InfluenceSelectedMessage args)
+    {
+        if (!_prototype.TryIndex(args.InfluenceProtoId, out var proto) || !TryComp<ActivatableUIComponent>(ent, out var uiComp) || !TryComp<CosmicCultComponent>(uiComp.CurrentSingleUser, out var cultComp))
+            return;
+        if (ent.Comp.AvailableEntropy < proto.Cost || cultComp.OwnedInfluences.Contains(proto) || uiComp.CurrentSingleUser == null)
+            return;
+        else cultComp.OwnedInfluences.Add(proto);
+
+        if (proto.InfluenceType == "influence-type-active")
+        {
+            var actionEnt = _actions.AddAction(uiComp.CurrentSingleUser.Value, proto.Action);
+            cultComp.ActionEntities.Add(actionEnt);
+        }
+        else if (proto.InfluenceType == "influence-type-passive")
+        {
+            UnlockPassive(uiComp.CurrentSingleUser.Value, proto); //Not unlocking an action? call the helper function to add the influence's passive effects
+        }
+
+        ent.Comp.AvailableEntropy -= proto.Cost;
+        cultComp.EntropyBudget -= proto.Cost;
+        ent.Comp.UnlockedInfluences.Remove(args.InfluenceProtoId);
+        cultComp.UnlockedInfluences.Remove(args.InfluenceProtoId);
+
+        _uiSystem.SetUiState(ent.Owner, MonumentKey.Key, GenerateBuiState(ent.Comp));
+    }
+    #endregion
+
+    #region Helper functions
+    private MonumentBuiState GenerateBuiState(MonumentComponent comp)
+    {
+        return new MonumentBuiState(
+            comp.AvailableEntropy,
+            comp.EntropyUntilNextStage,
+            comp.CrewToConvertNextStage,
+            comp.PercentageComplete,
+            comp.SelectedGlyph,
+            comp.UnlockedInfluences,
+            comp.UnlockedGlyphs
+        );
+    }
+    #endregion
+
+    private void UnlockPassive(EntityUid cultist, InfluencePrototype proto)
+    {
+        switch (proto.PassiveName) // Yay, switch statements.
+        {
+            case "eschew":
+                RemCompDeferred<HungerComponent>(cultist);
+                RemCompDeferred<ThirstComponent>(cultist);
+                break;
+            case "step":
+                EnsureComp<MovementIgnoreGravityComponent>(cultist);
+                break;
+            case "stride":
+                EnsureComp<InfluenceStrideComponent>(cultist);
+                break;
+            case "vitality":
+                EnsureComp<InfluenceVitalityComponent>(cultist);
+                break;
+            default:
+                break;
+        }
+    }
```

Comment:
Instead of hardcoding the component effects, make them settable via yaml probably